core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -140,47 +155,59 @@ private SingleThreaded(InternalDriverContext context) {
               .getDefaultProfile()
               .getDuration(DefaultDriverOption.CONFIG_RELOAD_INTERVAL);
 
-      forceLoadListenerKey =
-          this.eventBus.register(
-              ForceReloadConfigEvent.class, e -> RunOrSchedule.on(adminExecutor, this::reload));
-
-      RunOrSchedule.on(adminExecutor, this::scheduleReloadTask);
+      RunOrSchedule.on(adminExecutor, this::schedulePeriodicReload);
     }
 
-    private void scheduleReloadTask() {
+    private void schedulePeriodicReload() {
       assert adminExecutor.inEventLoop();
       // Cancel any previously running task
-      if (reloadFuture != null) {
-        reloadFuture.cancel(false);
+      if (periodicTaskHandle != null) {
+        periodicTaskHandle.cancel(false);
       }
       if (reloadInterval.isZero()) {
         LOG.debug("[{}] Reload interval is 0, disabling periodic reloading", logPrefix);
       } else {
         LOG.debug("[{}] Scheduling periodic reloading with interval {}", logPrefix, reloadInterval);
-        reloadFuture =
+        periodicTaskHandle =
             adminExecutor.scheduleAtFixedRate(
-                this::reload,
+                () -> reload(null),
                 reloadInterval.toNanos(),
                 reloadInterval.toNanos(),
                 TimeUnit.NANOSECONDS);
       }
     }
 
-    private void reload() {
+    private void reload(CompletableFuture<Void> reloadedFuture) {
       assert adminExecutor.inEventLoop();
       if (closeWasCalled) {
+        if (reloadedFuture != null) {
+          reloadedFuture.completeExceptionally(new IllegalStateException("session is closing"));
+        }
         return;
       }
-      if (driverConfig.reload(configSupplier.get())) {
-        LOG.info("[{}] Detected a configuration change", logPrefix);
-        eventBus.fire(ConfigChangeEvent.INSTANCE);
-        Duration newReloadInterval = config.getDuration(DefaultDriverOption.CONFIG_RELOAD_INTERVAL);
-        if (!newReloadInterval.equals(reloadInterval)) {
-          reloadInterval = newReloadInterval;
-          scheduleReloadTask();
+      try {
+        if (driverConfig.reload(configSupplier.get())) {
+          LOG.info("[{}] Detected a configuration change", logPrefix);
+          eventBus.fire(ConfigChangeEvent.INSTANCE);
```

Comment:
could you tell me where that signal is used on the consumer side? I was not able to find that

Reply:
`MetadataManager.onConfigChanged()` and `ChannelPool.onConfigChanged()`. It's also intended for future 3rd-party components.

The trick is that `eventBus.register()` takes a `Class`, not an object, so it does not reference `INSTANCE`. I see how it can be deceptive so I'll add a comment in the enum.